src/features/defi/providers/osmosis/components/OsmosisManager/Staking/Deposit/components/Deposit.tsx

```diff
@@ -157,15 +158,15 @@ export const Deposit: React.FC<DepositProps> = ({
   const handleCancel = history.goBack
 
   const validateCryptoAmount = (value: string) => {
-    const crypto = bnOrZero(balance).div(`1e+${asset.precision}`)
+    const crypto = amountAvailableCryptoPrecision
     const _value = bnOrZero(value)
     const hasValidBalance = crypto.gt(0) && _value.gt(0) && crypto.gte(value)
     if (_value.isEqualTo(0)) return ''
     return hasValidBalance || 'common.insufficientFunds'
   }
 
   const validateFiatAmount = (value: string) => {
-    const crypto = bnOrZero(balance).div(`1e+${asset.precision}`)
+    const crypto = amountAvailableCryptoPrecision
```

Comment:
Note, this doesn't solve anything yet, manual input is still rugged.

@pastaghost :

Why do we have to use a single-purpose `opportunityBalances?.fiatBalance` ? The total fiat balance of the underlyings should be equal to the fiat balance of the LP token.
However, both the LP price and the balance are currently wrong. These need to be fixed at portfolio/marketData level, not monkey-patched at view-layer.

Reply:
@gomesalexandre 

We don't anymore. 
I put that in before I moved the Osmosis LP asset price calculation logic down into market-service and missed that line when cleaning up. `opportunityBalances.fiatBalance` can be removed.